### PR TITLE
Rename fenicsadapter to fenicsprecice.

### LINKED ISSUE #85: Naming of software, package and repository
We want to use the following names for this project:

* The name of the software is "FEniCS-preCICE Adapter", since this is a name that tells both preCICE and FEniCS users/developers what the software does.
* The name of this repository is `precice/fenics-adapter`, since this is consistent with the other adapters and represents the perspective of a preCICE developer.
* The name of the package is `fenicsprecice`, since this repesents the perspective of a FEniCS user `import fenicsprecice`.

This comes with a few todos:

* Documentation (`README.md`)
* Related #75 
* Update `setup.py`
* Change `import fenicsadapter` to `import fenicsprecice` in corresponding tutorials (e.g. https://github.com/precice/tutorials/blob/master/HT/partitioned-heat/fenics-fenics/heat.py#L30)

### PR DESCRIPTION
**Pull request by BenjaminRueth**
closes #85 

@IshaanDesai , @uekerman : I hope that I did not miss any place, where renaming is required.
@IshaanDesai: Please veto, if this renaming conflicts too much with other open PRs. The renaming can also wait until #71 is merged, if you prefer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# fenics-adapter
+# preCICE-adapter for the open source computing platform FEniCS
```
Comment (uekerman) [inline README.md:1]:
```suggestion
# FEniCS-preCICE adapter
```

**Review by IshaanDesai — APPROVED** [review]
As there are no changes to files which actually have adapter functionality I think this can be merged to develop. The name change will be done in https://github.com/precice/fenics-adapter/pull/71 manually.

File: setup.py [hunk 2]
```diff
@@ -3,21 +3,21 @@
...
 version = {}
-with open("fenicsadapter/version.py") as fp:
+with open("fenicsprecice/version.py") as fp:
     exec(fp.read(), version)
 
-setup(name='fenicsadapter',
+setup(name='fenicsprecice',
       version=version['__version__'],
-      description='preCICE-adapter for the open source computing platform FEniCS ',
+      description='preCICE-adapter for the open source computing platform FEniCS',
```
Comment (BenjaminRueth) [inline setup.py:15]:
```suggestion
      description='FEniCS-preCICE adapter: A preCICE-adapter for the open source computing platform FEniCS.',
```

Comment (BenjaminRueth) [reply setup.py:15]: Following the suggestion in https://github.com/precice/fenics-adapter/pull/93#discussion_r510771521, I would use this `description`.

**Comment by BenjaminRueth** [comment]
@uekerman Does https://github.com/precice/fenics-adapter/pull/93/commits/7841946013998bbe90f143116ebf98cb81c05944 look ok?

File: setup.py [hunk 3]
```diff
@@ -3,21 +3,21 @@
 # from https://stackoverflow.com/a/9079062
 import sys  
 if sys.version_info[0] < 3:
-    raise Exception("The fenicsadapter only supports Python3. Did you run $python setup.py <option>.? Try running $python3 setup.py <option>.")
+    raise Exception("The fenicsprecice only supports Python3. Did you run $python setup.py <option>.? Try running $python3 setup.py <option>.")
```
Comment (BenjaminRueth) [inline setup.py:6]:
```suggestion
    raise Exception("fenicsprecice only supports Python3. Did you run $python setup.py <option>.? Try running $python3 setup.py <option>.")
```

File: setup.py [hunk 4]
```diff
@@ -3,21 +3,21 @@
...
 version = {}
-with open("fenicsadapter/version.py") as fp:
+with open("fenicsprecice/version.py") as fp:
     exec(fp.read(), version)
 
-setup(name='fenicsadapter',
+setup(name='fenicsprecice',
       version=version['__version__'],
-      description='preCICE-adapter for the open source computing platform FEniCS ',
+      description='FEniCS-preCICE adapter: A preCICE-adapter for the open source computing platform FEniCS.',
```
Comment (BenjaminRueth) [inline setup.py:15]:
```suggestion
      description='FEniCS-preCICE adapter is a preCICE adapter for the open source computing platform FEniCS.',
```
I checked what numpy is doing here:
```
$ pip3 show numpy
Name: numpy
Version: 1.17.4
Summary: NumPy is the fundamental package for array computing with Python.
Home-page: https://www.numpy.org
...
```
They use the name of the software (including "complicated" typography) in the `description` (Summary). Applied to our software this means we should use the official name "FEniCS-preCICE adapter" here:
```
$ pip3 show fenicsprecice
Name: fenicsprecice
Version: 0.2
Summary: FEniCS-preCICE adapter is a preCICE adapter for the open source computing platform FEniCS.
Home-page: https://github.com/precice/fenics-adapter
...
```
Does this look good?

**Comment by BenjaminRueth** [comment]
I will merge this PR now. Note that the tutorials will need a (minor) Update, as well. I will take care of this in a moment.

**Pull request merged**